pangeo_forge_recipes/executors/base.py

```diff
@@ -27,12 +27,27 @@ class Stage:
     mappable: Optional[Iterable] = None
     annotations: Optional[StageAnnotations] = None
 
+    @property
+    def ismappable(self):
+        return True if self.mappable is not None else False
+
 
 @dataclass(frozen=True)
 class Pipeline:
     stages: Iterable[Stage]
     config: Optional[Config] = None
 
+    def __iter__(self):
+        for stage in self.stages:
+            yield stage.name
+
+    def __getitem__(self, name):
+        names = [s.name for s in self.stages]
+        if name not in names:
+            raise KeyError(f"'{name}' not a stage name in this pipeline. Must be one of {names}.")
+        stage = [s for s in self.stages if s.name == name][0]
+        return stage
```

Comment:
I don't see why it's necessary to add dictionary semantics to the Pipeline class. Is it really so much simpler to say
```python
for stage in pipeline:
    ...
```
instead of
```python
for stage in pipeline.stages:
    ...
```

The iterator here also returns only the stage name, rather than the stage itself. That necessitates the additional `__getitem__` method, which is also unnecessary if you just iterate through `pipeline.stages` directly.

What I would instead recommend is to provide custom and verbose `__repr__` and [`_repr_html`](https://ipython.readthedocs.io/en/stable/config/integrating.html#rich-display) methods on the class. These, together with a better docstring, should give the user all the information they need to explore the pipeline.